perllib/FixMyStreet/Map/Buckinghamshire.pm

```diff
@@ -2,19 +2,88 @@
 # More JavaScript, for street assets
 
 package FixMyStreet::Map::Buckinghamshire;
-use base 'FixMyStreet::Map::OSM';
+use base 'FixMyStreet::Map::WMTSBase';
 
 use strict;
 
+sub zoom_parameters {
+    my $self = shift;
+    my $params = {
+        zoom_levels    => scalar $self->scales,
+        default_zoom   => 5,
+        min_zoom_level => 0,
+        id_offset      => 0,
+    };
+    return $params;
+}
+
+sub tile_parameters {
+    my $self = shift;
+    my $params = {
+        urls            => [ 'https://maps.buckscc.gov.uk/arcgis/rest/services/Basemapping2018/MapServer/WMTS/tile' ],
+        layer_names     => [ 'Basemapping2018' ],
+        wmts_version    => '1.0.0',
+        layer_style     => 'default',
+        matrix_set      => 'default028mm',
+        suffix          => '.png', # appended to tile URLs
+        size            => 256, # pixels
+        dpi             => 96,
+        inches_per_unit => 39.3701, # BNG uses metres
+        projection      => 'EPSG:27700',
+        origin_x        => -5220400.0,
+        origin_y        => 4470200.0,
+    };
+    return $params;
+}
+
+sub scales {
+    my $self = shift;
+    my @scales = (
+        '1000000',
+        '500000',
+        '250000',
+        '125000',
+        '64000',
+        '32000',
+        '16000',
+        '8000',
+        '4000',
+        '2000',
+        '1000',
+    );
+    return @scales;
+
+}
+
+sub copyright {
+    return '&copy; BCC';
+}
+
+sub map_template { 'buckinghamshire' }
+
 sub map_javascript { [
-    '/vendor/OpenLayers/OpenLayers.wfs.js',
-    '/vendor/OpenLayers.Projection.OrdnanceSurvey.js',
+    '/vendor/OpenLayers/OpenLayers.bristol.js',
```

Comment:
Could/should this file be renamed to something not Bristol-specific now it's being shared? (And same for `/data/bristol/openlayers.bristol.cfg`)